llama_index/vector_stores/types.py

```diff
@@ -65,23 +65,47 @@ class ExactMatchFilter(BaseModel):
     value: Union[StrictInt, StrictFloat, StrictStr]
 
 
+class AdvancedFilter(BaseModel):
+    """Advanced metadata filter for vector stores to support more operators.
+
+    Value uses Strict* types, as int, float and str are compatible types and were all
+    converted to string before.
+
+    See: https://docs.pydantic.dev/latest/usage/types/#strict-types
+    """
+
+    key: str
+    value: Union[StrictInt, StrictFloat, StrictStr]
+    operator: Optional[str] = None
+
+
 class MetadataFilters(BaseModel):
     """Metadata filters for vector stores.
 
     Currently only supports exact match filters.
     TODO: support more advanced expressions.
     """
 
-    filters: List[ExactMatchFilter]
+    # Exact match filters
+    filters: Optional[List[ExactMatchFilter]] = None
+    # Advanced filters with operators like >, <, >=, <=, !=, etc.
+    advanced_filters: Optional[List[AdvancedFilter]] = None
```

Comment:
How does support for `and` or `or` work with filters? I guess these abstractions don't quite capture that yet

Reply:
pass the `condition` as well as part of `MetadataFilters`

Reply:
Does that support more complicated filters?

i.e. `(key1 == val1 or key1 == val2) and key2 == val3`

Reply:
(Just checking -- this doesn't have to be in this PR if its not supported, but just thinking in the future)

Reply:
yeah it will be next step. nested conditions could be tricky and it should be dependent on users